polynote-kernel/src/main/scala/polynote/kernel/interpreter/scal/ScalaInterpreter.scala

```diff
@@ -39,14 +39,14 @@ class ScalaInterpreter private[scal] (
   override def completionsAt(code: String, pos: Int, state: State): RIO[Blocking, List[Completion]] = for {
     collectedState   <- injectState(collectState(state)).provideLayer(CurrentRuntime.noRuntime)
     valDefs           = collectedState.values.mapValues(_._1).values.toList
-    cellCode         <- scalaCompiler.cellCode(s"Cell${state.id.toString}", s"\n\n${code.substring(0, math.min(pos, code.length))}  ", collectedState.prevCells, valDefs, collectedState.imports, strictParse = false)
+    cellCode         <- scalaCompiler.cellCode(state.id.toString, s"\n\n${code.substring(0, math.min(pos, code.length))}  ", collectedState.prevCells, valDefs, collectedState.imports, strictParse = false)
     completions      <- completer.completions(cellCode, pos + 2)
   } yield completions
 
   override def parametersAt(code: String, pos: Int, state: State): RIO[Blocking, Option[Signatures]] = for {
     collectedState <- injectState(collectState(state)).provideLayer(CurrentRuntime.noRuntime)
     valDefs         = collectedState.values.mapValues(_._1).values.toList
-    cellCode       <- scalaCompiler.cellCode(s"Cell${state.id.toString}", s"\n\n$code  ", collectedState.prevCells, valDefs, collectedState.imports, strictParse = false)
```

Comment:
Maybe I just missed it in testing, but I didn't see a reason to keep the ID as a string formatted like this (and it makes lookups on the client-side harder as you have to parse the string for a number). Thought this would be easier, so I refactored this in a few places.

Reply:
I think we were hoping to eventually move away from the monotonic numbering as the identifier, and use UUIDs or something instead.